editor/components/page-attributes/template.js

```diff
@@ -42,27 +40,19 @@ export function PageTemplate( { availableTemplates, selectedTemplate, instanceId
 	);
 }
 
-const applyConnect = connect(
-	( state ) => {
+export default compose(
+	withSelect( ( select ) => {
 		return {
-			selectedTemplate: getEditedPostAttribute( state, 'template' ),
+			selectedTemplate: select( 'core/editor' ).getEditedPostAttribute( 'template' ),
 		};
-	},
-	{
+	} ),
+	withDispatch( ( dispatch ) => ( {
 		onUpdate( templateSlug ) {
-			return editPost( { template: templateSlug || '' } );
+			dispatch( 'core/editor' ).editPost( { template: templateSlug || '' } );
 		},
-	}
-);
-
-const applyWithEditorSettings = withEditorSettings(
-	( settings ) => ( {
+	} ) ),
+	withEditorSettings( ( settings ) => ( {
```

Comment:
Minor: Normally I prefer to have withEditorSettings before withSelect and withDispatch as what we query from the store may one day depend on the editor settings but the editor settings don't depend on the store.

Reply:
Well, it depends sometimes you need the props (that can come from the store) to check or not a setting. But I guess it's less common

Reply:
Yes valid point, I had only seen the first case so it did not occur to me the second case could also be valid.